### PR TITLE
fix: resolve tilde in sasjs config paths

### LINKED ISSUE #794: sasjs compile does not recognise home directory (`~`)
When using a tilde (`~`) in the macroFolders array, the compile process thinks this is a relative (to the sasjs project) path

image

Expected behaviour - if the path starts with a `~` it should be considered an absolute path

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Closes #794 

## Intent

resolve paths with `~` in sasjsConfig.

## Implementation

Added `getAbsolutePath` to `utils/utils.ts`, and extracted logic from `commands/*`

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked:
image
Server:
image
- [x] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

**Review by allanbowe — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
   },
   "dependencies": {
     "@sasjs/adapter": "2.8.9",
-    "@sasjs/core": "2.35.3",
+    "@sasjs/core": "^2.35.4",
```
Comment (krishna-acondy) [inline package.json:54]: Is the `^` intentional?

File: src/commands/compile/spec/compile.spec.ts [hunk 2]
```diff
@@ -76,6 +80,17 @@ describe('sasjs compile', () => {
     expect(compileModule.compileJobsServicesTests).toHaveBeenCalled()
   })
 
+  it('should compile project with having tilda in paths', async () => {
```
Comment (krishna-acondy) [inline src/commands/compile/spec/compile.spec.ts:83]: `tilde`.

File: src/utils/utils.ts [hunk 3]
```diff
@@ -393,3 +394,8 @@ parmcards4;
     `Please deploy the SASjs runner by running the code below and try again:\n${sasjsRunnerCode}`
   )
 }
+
+export const getAbsolutePath = (providedPath: string, relativePath: string) =>
```
Comment (krishna-acondy) [inline src/utils/utils.ts:398]: It would be good to see some tests for this.

File: package.json (hunk 1 above)
Comment (saadjutt01) [reply package.json:54]: corrected

File: src/commands/compile/spec/compile.spec.ts (hunk 2 above)
Comment (saadjutt01) [reply src/commands/compile/spec/compile.spec.ts:83]: corrected

File: src/utils/utils.ts (hunk 3 above)
Comment (saadjutt01) [reply src/utils/utils.ts:398]: Added

**Review by VladislavParhomchik — DISMISSED** [review]

**Review by VladislavParhomchik — APPROVED** [review]

**Pull request merged**